integration/test/Test/Conversation.hs

```diff
@@ -135,3 +136,43 @@ testFederationStatus = do
     $ \resp -> do
       resp.status `shouldMatchInt` 400
       resp.json %. "label" `shouldMatch` "discovery-failure"
+
+testCreateConversationFullyConnected :: HasCallStack => App ()
+testCreateConversationFullyConnected = do
+  let setFederationConfig =
+        setField "optSettings.setFederationStrategy" "allowDynamic"
+          >=> removeField "optSettings.setFederationDomainConfigs"
+          >=> setField "optSettings.setFederationDomainConfigsUpdateFreq" (Aeson.Number 1)
+  startDynamicBackends
+    [ def {dbBrig = setFederationConfig},
+      def {dbBrig = setFederationConfig},
+      def {dbBrig = setFederationConfig}
+    ]
+    $ \dynDomains -> do
+      domains@[domainA, domainB, domainC] <- pure dynDomains
+      sequence_ [Internal.createFedConn x (Internal.FedConn y "full_search") | x <- domains, y <- domains]
```

Comment:
```suggestion
      sequence_ [Internal.createFedConn x (Internal.FedConn y "full_search") | x <- domains, y <- domains, x /= y]
```

(it works without this change, but it's weird to register a domain to federate with itself.)